src/server/game/Loot/LootMgr.cpp

```diff
@@ -445,6 +445,35 @@ void Loot::AddItem(LootStoreItem const& item)
         lootItems.push_back(generatedLoot);
         count -= proto->GetMaxStackSize();
 
+        // In some cases, a dropped item should be visible/lootable only for some players in group
+        bool canSeeItemInLootWindow = false;
+        if (auto player = ObjectAccessor::FindPlayer(lootOwnerGUID))
+        {
+            if (auto group = player->GetGroup())
+            {
+                for (auto itr = group->GetFirstMember(); itr != nullptr; itr = itr->next())
+                {
+                    if (auto member = itr->GetSource())
+                    {
+                        if (generatedLoot.AllowedForPlayer(member))
+                        {
+                            canSeeItemInLootWindow = true;
+                            break;
```

Comment:
@locus313 I've added a few minor improvements to the code. However, I still don't get the whole meaning of this logic. We first check the player and then each member of the party with `AllowedForPlayer` . If we find at least 1 member of the party, or the player itself, we set `canSeeItemInLootWindow = true;` otherwise it will be `false`. I don't see how this can fix the issue but I haven't dug much into this problem